File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizMonthPickerIsland.swift
import SwiftUI

/**
 * BizMonthPickerIsland - SwiftUI component matching Web UI-Framework `PPMonthPickerIsland.vue`.
 */
public struct BizMonthPickerIsland: View {
    public var modelValue: Bool
    public var title: String
    public var config: String
    public var initialDate: String
    public var showActionButtons: Bool
    public var cancelText: String
    public var confirmText: String
    public var position: String
    public var fullWidth: Bool
    public var offset: Int
    public var action: () -> Void

    public init(
        modelValue: Bool = false,
        title: String = "",
        config: String = "",
        initialDate: String = "",
        showActionButtons: Bool = false,
        cancelText: String = "",
        confirmText: String = "",
        position: String = "top",
        fullWidth: Bool = false,
        offset: Int = 16,
        action: @escaping () -> Void = {}
    ) {
        self.modelValue = modelValue
        self.title = title
        self.config = config
        self.initialDate = initialDate
        self.showActionButtons = showActionButtons
        self.cancelText = cancelText
        self.confirmText = confirmText
        self.position = position
        self.fullWidth = fullWidth
        self.offset = offset
        self.action = action
    }

    public var body: some View {
        Button(action: action) {
            VStack(alignment: .leading, spacing: 8) {
                Text("MonthPickerIsland")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(Color(hex: "#1A2A5E"))
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(16)
            .background(Color.white)
            .cornerRadius(12)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(Color(hex: "#CBD5E1"), lineWidth: 1)
            )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

/**
 * Web UI-Framework parity alias for [BizMonthPickerIsland].
 */
public typealias PPMonthPickerIsland = BizMonthPickerIsland

File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizMonthPickerIsland.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * BizMonthPickerIsland - Jetpack Compose component matching Web UI-Framework `PPMonthPickerIsland.vue`.
 */
@Composable
fun BizMonthPickerIsland(
    modifier: Modifier = Modifier,
    modelValue: Boolean = false,
    title: String = "",
    config: String = "",
    initialDate: String = "",
    showActionButtons: Boolean = false,
    cancelText: String = "",
    confirmText: String = "",
    position: String = "top",
    fullWidth: Boolean = false,
    offset: Int = 16,
    onClick: () -> Unit = {},
    content: @Composable (() -> Unit)? = null
) {
    val primaryNavy = Color(0xFF1A2A5E)
    val primaryBlue = Color(0xFF003399)
    val secondaryBg = Color(0xFFF1F5F9)
    val borderColor = Color(0xFFCBD5E1)

    Box(
        modifier = modifier
            .fillMaxWidth()
            .defaultMinSize(minHeight = 44.dp)
            .background(Color.White, shape = RoundedCornerShape(12.dp))
            .border(1.dp, borderColor, shape = RoundedCornerShape(12.dp))
            .padding(16.dp)
            .clickable(onClick = onClick),
        contentAlignment = Alignment.CenterStart
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween,
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = "MonthPickerIsland",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = primaryNavy
                )
                if (content != null) {
                    Spacer(modifier = Modifier.height(8.dp))
                    content()
                }
            }
        }
    }
}

/**
 * Web UI-Framework parity alias for [BizMonthPickerIsland].
 */
@Composable
fun PPMonthPickerIsland(
    modifier: Modifier = Modifier,
    modelValue: Boolean = false,
    title: String = "",
    config: String = "",
    initialDate: String = "",
    showActionButtons: Boolean = false,
    cancelText: String = "",
    confirmText: String = "",
    position: String = "top",
    fullWidth: Boolean = false,
    offset: Int = 16,
    onClick: () -> Unit = {},
    content: @Composable (() -> Unit)? = null
) = BizMonthPickerIsland(
        modifier = modifier,
        modelValue = modelValue,
        title = title,
        config = config,
        initialDate = initialDate,
        showActionButtons = showActionButtons,
        cancelText = cancelText,
        confirmText = confirmText,
        position = position,
        fullWidth = fullWidth,
        offset = offset,
        onClick = onClick,
        content = content
)
